subsys/net/ip/net_if.c

```diff
@@ -2848,7 +2848,7 @@ int net_if_up(struct net_if *iface)
 
 #if defined(CONFIG_NET_OFFLOAD)
 	if (net_if_is_ip_offloaded(iface)) {
-		goto done;
+		goto exit;
```

Comment:
I think you'll need to rework the code below so you'll get net_if_flags_set(iface, NET_IF_UP) being called appropriately

Reply:
Good point, I've missed the `net_if_flag_set`, focusing only on `net_mgmt_event_notify`, will do.

Reply:
I've added an extra call to `atomic_set_bit` for offloaded interface, as I found the code overcomplicated after modifying the code below `done` label. Let me know if it's ok.

Reply:
use net_if_flag_set() instead of atomic_set_bit() but yeah imo that's the way to go

Reply:
Right, I've used an older code base, hece copied `atomic_set_bit`, will fix